okhttp/src/main/kotlin/okhttp3/internal/platform/Android10Platform.kt

```diff
@@ -81,6 +87,26 @@ class Android10Platform : Platform() {
   override fun buildCertificateChainCleaner(trustManager: X509TrustManager): CertificateChainCleaner =
       AndroidCertificateChainCleaner.buildIfSupported(trustManager) ?: super.buildCertificateChainCleaner(trustManager)
 
+  override fun IDNtoASCII(domain: String): String {
+    val output = StringBuilder()
+    val info = IDNA.Info()
+    idna.nameToASCII(domain, output, info)
+    if (info.hasErrors()) {
```

Comment:
Should we throw on info.hasErrors? Or conditionally based on the errors?

Reply:
You should throw when there is an error. As far as I know ICU4j correctly handles valid labels.